include/aws/common/random_access_set.h

```diff
@@ -0,0 +1,80 @@
+#ifndef AWS_COMMON_RANDOM_ACCESS_SET_H
+#define AWS_COMMON_RANDOM_ACCESS_SET_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/array_list.h>
+#include <aws/common/common.h>
+#include <aws/common/hash_table.h>
+
+struct aws_random_access_set_impl;
+
+struct aws_random_access_set {
+    struct aws_random_access_set_impl *impl;
+};
+
+AWS_EXTERN_C_BEGIN
+
+/* TODO: CBMC verification. */
```

Comment:
Do we need these to merge to main?

Reply:
We should probably address that before merge... But, probably better to have a separate PR. I don't want to prioritize it before the stream manager thought...